packages/firestore/src/local/simple_db.ts

```diff
@@ -578,7 +599,9 @@ export class SimpleDbStore<
     const cursorRequest = this.cursor({});
     return new PersistencePromise((resolve, reject) => {
       cursorRequest.onerror = (event: Event) => {
-        reject((event.target as IDBRequest).error!);
+        const error = (event.target as IDBRequest).error!;
```

Comment:
You could move this cast into `checkForAndReportiOSError`, which could then do an `instanceof` check..

Reply:
we need the cast result for `reject(error)` too though...  and IDBRequest is an interface not a class, so no instanceof checks. :-/